tests/migration/eviction_strategy.go

```diff
@@ -0,0 +1,722 @@
+
+	"kubevirt.io/kubevirt/tests/libwait"
+
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+)
+
+var _ = SIGMigrationDescribe("Live Migration", func() {
+	var (
+		virtClient kubecli.KubevirtClient
+		err        error
+	)
+
+	BeforeEach(func() {
+		checks.SkipIfMigrationIsNotPossible()
+		virtClient = kubevirt.Client()
+	})
+
+	setControlPlaneUnschedulable := func(mode bool) {
+		controlPlaneNodes, err := virtClient.
+			CoreV1().
+			Nodes().
+			List(context.Background(),
+				metav1.ListOptions{LabelSelector: `node-role.kubernetes.io/control-plane`})
+		Expect(err).ShouldNot(HaveOccurred(), "could not list control-plane nodes")
+		Expect(controlPlaneNodes.Items).ShouldNot(BeEmpty(),
+			"There are no control-plane nodes in the cluster")
+
+		for _, node := range controlPlaneNodes.Items {
+			if node.Spec.Unschedulable == mode {
+				continue
+			}
+
+			nodeCopy := node.DeepCopy()
+			nodeCopy.Spec.Unschedulable = mode
+
+			oldData, err := json.Marshal(node)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			newData, err := json.Marshal(nodeCopy)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			patch, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, node)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			_, err = virtClient.CoreV1().Nodes().Patch(context.Background(), node.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+		}
+	}
+
+	// temporaryNodeDrain also sets the `NoSchedule` taint on the node.
+	// nodes with this taint will be reset to their original state on each
+	// test teardown by the test framework. Check `libnode.CleanNodes`.
+	temporaryNodeDrain := func(nodeName string) {
+		By("taining the node with `NoExecute`, the framework will reset the node's taints and un-schedulable properties on test teardown")
+		libnode.Taint(nodeName, libnode.GetNodeDrainKey(), k8sv1.TaintEffectNoSchedule)
+
+		By(fmt.Sprintf("Draining node %s", nodeName))
+		// we can't really expect an error during node drain because vms with eviction strategy can be migrated by the
+		// time that we call it.
+		vmiSelector := v1.AppLabel + "=virt-launcher"
+		k8sClient := clientcmd.GetK8sCmdClient()
+		if k8sClient == "oc" {
+			_, _, err := clientcmd.RunCommandWithNS("", k8sClient, "adm", "drain", nodeName, "--delete-emptydir-data", "--pod-selector", vmiSelector,
+				"--ignore-daemonsets=true", "--force", "--timeout=180s")
+			Expect(err).ToNot(HaveOccurred())
+		} else {
+			_, _, err := clientcmd.RunCommandWithNS("", k8sClient, "drain", nodeName, "--delete-emptydir-data", "--pod-selector", vmiSelector,
+				"--ignore-daemonsets=true", "--force", "--timeout=180s")
+			Expect(err).ToNot(HaveOccurred())
+		}
+	}
+
+	Context("with a live-migrate eviction strategy set", func() {
```

Comment:
Since we have several nested `Context` and `Describe` nodes here, that might indicate that we should split into even smaller files. WDYT?

I.e. this one could go into `live-migrate-eviction-strategy.go`

Reply:
Thanks Daniel. IMHO I think that this split will create too much granularity that we don't need right now. Consider that there are only two tests in `cluster-wide-live-migrate-eviction-strategy`, so the value added would be minimal. Nothing prohibits us from doing it in the future in case we need it. Let me know!